src/lib/operators/join_sort_merge/column_materializer.hpp

```diff
@@ -84,44 +58,56 @@ class ColumnMaterializer : public ColumnVisitable {
     return output;
   }
 
+ private:
   /**
-  * ColumnVisitable implementation to materialize and sort a value column.
-  **/
-  void handle_value_column(const BaseValueColumn& column, std::shared_ptr<ColumnVisitableContext> context) override {
-    auto& value_column = static_cast<const ValueColumn<T>&>(column);
-    auto materialization_context = std::static_pointer_cast<MaterializationContext>(context);
-    auto output = std::make_shared<MaterializedColumn<T>>();
-    output->reserve(value_column.values().size());
-
-    // Copy over every entry
-    for (ChunkOffset chunk_offset{0}; chunk_offset < value_column.values().size(); ++chunk_offset) {
-      RowID row_id{materialization_context->chunk_id, chunk_offset};
+   * Creates a job to materialize and sort a chunk.
+   **/
+  std::shared_ptr<JobTask> _create_chunk_materialization_job(std::unique_ptr<MaterializedColumnList<T>>& output,
```

Comment:
@mjendruk @mrks Maybe this is obvious to you and I know this is just c/p code but anyway:

We're using a **reference** to a unique_ptr here, then **capture** that reference by reference and use it in a deferred lambda and the only reason we're save here is because the one and only caller of this method calls `CurrentScheduler::wait_for_tasks(jobs)` before the unique_ptr goes out of scope.

The fact that this made it through its original review and the c/p in this PR tells me (at least) that we're making a decent choice when making stack instances and shared_ptr the default ways to store objects.

Reply:
We are NOT safe here. If one of the threads throws an exception, the calling stack gets destroyed and the remaining threads will segfault. That's already on my list and I fixed it for a couple of instances in #541.

Reply:
holy hell.

Reply:
This is why the new (https://github.com/hyrise/hyrise/wiki/Code-Style)[flowcharts] require you to own everything that you pass into a thread.

Reply:
> the only reason we're save here is because the one and only caller of this method calls CurrentScheduler::wait_for_tasks(jobs) before the unique_ptr goes out of scope.

Yes, that’s why I didn’t bother changes anything except the actual materialization code. I didn’t understand the reasoning behind the usage of shared pointers vs. unique pointers here, so I left it as it was.

> We are NOT safe here. If one of the threads throws an exception, the calling stack gets destroyed and the remaining threads will segfault. That's already on my list and I fixed it for a couple of instances in #541.

Could you elaborate on this? How does an exception in one thread affect another?

Reply:
About the pointers:  unique pointers are used because there is only one owner of the pointer, which is the column materializer instance. Once the materialization finishes the unique pointer is "given away" (std::move) to the user of the materializer. The caller receives a unique pointer and thus can be sure that he is the only owner of that pointer. Therefore the caller knows he can be the only one who has access to the instance and who is responsible of the time at which the data structure behind the pointer is freed.  The receiver has 100% control.

Reply:
>> We are NOT safe here. If one of the threads throws an exception, the calling stack gets destroyed and the remaining threads will segfault. That's already on my list and I fixed it for a couple of instances in #541.

> Could you elaborate on this? How does an exception in one thread affect another?

Never mind - misunderstanding on my side.